src/libraries/System.Console/src/System/TermInfo.cs

```diff
@@ -264,9 +264,13 @@ private static bool TryOpen(string filePath, [NotNullWhen(true)] out SafeFileHan
                     int fileLen = (int)termInfoLength;
 
                     byte[] data = new byte[fileLen];
-                    if (ConsolePal.Read(fd, data, 0, fileLen) != fileLen)
+                    fixed (byte* bufPtr = data)
```

Comment:
`ConsolePal.Read` causes the terminal to be configured since https://github.com/dotnet/runtime/pull/37491.
We can just read the file here.